src/Arrays/AppendedArrays.jl

```diff
@@ -63,7 +63,9 @@ struct AppendedArray{T,A,B} <: AbstractVector{T}
     B = typeof(b)
     ai = testitem(a)
     bi = testitem(b)
-    T = eltype(collect((ai,bi)))
+    #T = eltype(collect((ai,bi)))
+    #new{T,A,B}(a,b)
+    T = Union{typeof(ai),typeof(bi)}
```

Comment:
This would make the array type in-stable. Unfortunately, this is a performance critical part, and we want type stability here. I guess that the problem is that you want to have some part of the array with dual numbers, an the other part with regular ones, right?

Reply:
"you want to have some part of the array with dual numbers, an the other part with regular ones, right?"  Yes that is the object we have to deal with but since its eltype is not fixed it is not concrete and causes problems down the line.

Reply:
> This would make the array type in-stable. 

@fverdugo ... I have double checked that `Union{typeof(ai),typeof(bi)}` when `typeof(ai)==typeof(bi)` is equivalent to `typeof(ai)`. Why do you think that this addition of @ConnorMallon introduces unexisting type instability? Anyways, we still have to understand the problem and the cause behind this need of @ConnorMallon.